File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
       int zeros = 0;
       int l = 0, r = 0; 
       int n = nums.size();
       int ans = 0;
       
       while (r < n) { 
            if (nums[r] == 0) {
                zeros++;
            }
            
            while (zeros > k) {
                if (nums[l] == 0) {
                    zeros -= 1;
                }
                l++;
            }
            
            ans = std::max(ans, r - l + 1);
            r++;
       }
       
       return ans;
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        zeros = 0
        l = 0 
        r = 0
        n = len(nums)
        ans = 0
        while r <= n - 1:
            
            if nums[r] == 0:
                zeros += 1
            while zeros > k:
                if nums[l] == 0:
                    zeros -= 1
                l += 1
            ans = max(ans, r - l + 1)
            r += 1
        return ans 